lib/AST/Decl.cpp

```diff
@@ -1685,13 +1685,35 @@ bool swift::conflicting(const OverloadSignature& sig1,
   // If one is a compound name and the other is not, they do not conflict
   // if one is a property and the other is a non-nullary function.
   if (sig1.Name.isCompoundName() != sig2.Name.isCompoundName()) {
-    return !((sig1.IsProperty && !sig2.Name.getArgumentNames().empty()) ||
-             (sig2.IsProperty && !sig1.Name.getArgumentNames().empty()));
+    return !((sig1.IsVariable && !sig2.Name.getArgumentNames().empty()) ||
+             (sig2.IsVariable && !sig1.Name.getArgumentNames().empty()));
   }
   
   return sig1.Name == sig2.Name;
 }
 
+bool swift::conflicting(const OverloadSignature& sig1, CanType sig1Type,
+                        const OverloadSignature& sig2, CanType sig2Type,
+                        bool skipProtocolExtensionCheck) {
+  // If the signatures don't conflict to begin with, we're done.
+  if (!conflicting(sig1, sig2, skipProtocolExtensionCheck))
+    return false;
+
+  // Variables always conflict with non-variables with the same signature.
+  // (e.g variables with zero argument functions, variables with type
```

Comment:
Variables do not conflict with zero-argument functions (and you've implemented them to not conflict, so that's okay).

Reply:
Why are these new checks here and not inside the original `conflicting`?

Reply:
@jrose-apple Variables do indeed conflict with zero argument functions (we have `mixed_redecl6` in the `overload.swift` test for that), e.g:

```
struct Foo {
  var i: Int { return 0 }
  func i() {} // error: Invalid redeclaration of 'i()'
}
```
---
These new checks are in this function as they're only relevant when comparing the overload types (they prevent us from saying that two decls don't conflict because the overload types differ).

To be honest, I'm not too mad on having this logic separated across three functions (the two `conflicting` overloads, and the logic that computes the overload types), but I couldn't really see a better way to do it given the two current use cases of the logic. Any ideas on how to better structure things are more than welcome :)

Reply:
> Variables do indeed conflict with zero argument functions – we have `mixed_redecl6` in the `overload.swift` test for that

Aargh, right. You're not allowed to declare them, even though you can end up with that effect by importing other modules.